scipy/_lib/_array_api.py

```diff
@@ -163,3 +164,37 @@ def copy(x, *, xp=None):
         xp = array_namespace(x)
 
     return as_xparray(x, copy=True, xp=xp)
+
+
+def cov(x, *, xp=None):
+    if xp is None:
+        xp = array_namespace(x)
+
+    X = copy(x, xp=xp)
+    dtype = xp.result_type(X, xp.float64)
+
+    X = atleast_nd(X, ndim=2, xp=xp)
+    X = xp.asarray(X, dtype=dtype)
+
+    avg = xp.mean(X, axis=1)
+    w_sum = xp.asarray(size(X) / size(avg), dtype=avg.dtype)
+    if w_sum.shape != avg.shape:
+        w_sum = copy(xp.broadcast_to(w_sum, avg.shape), xp=xp)
+
+    w_sum = w_sum[0]
+
+    fact = X.shape[1] - 1
+
+    if fact <= 0:
+        warnings.warn("Degrees of freedom <= 0 for slice",
+                      RuntimeWarning, stacklevel=2)
+        fact = 0.0
+
+    X -= avg[:, None]
+    X_T = X.T
+    if xp.isdtype(X_T.dtype, 'complex floating'):
+        X_T = xp.conj(X_T)
+    c = xp.matmul(X, X_T)
+    c *= 1 / fact
+    axis = tuple(index for index, value in enumerate(c.shape) if value == 1)
```

Comment:
this could have better readability:
`axes = tuple(axis for axis, length in enumerate(c.shape) if length == 1` then `axis=axes`
or something like that may be better.